storage/src/main/java/org/apache/kafka/server/log/remote/metadata/storage/RemotePartitionMetadataStore.java

```diff
@@ -153,7 +153,7 @@ private FileBasedRemoteLogMetadataCache getRemoteLogMetadataCache(TopicIdPartiti
         }
 
         if (!remoteLogMetadataCache.isInitialized()) {
-            throw new ResourceNotReadyException("Remote log metadata cache is not initialized for partition: " + topicIdPartition);
+            throw new ReplicaNotAvailableException("Remote log metadata cache is not initialized for partition: " + topicIdPartition);
```

Comment:
Could we add a comment here to explain why we throw this exception, ex:
`Throwing a retriable ReplicaNotAvailableException here to for clients retry. We can introduce a new more appropriate exception with a KIP in the future.`  WDYT?

Reply:
Sure, I will add.